ext/opentelemetry-ext-mysql/tests/test_mysql_integration.py

```diff
@@ -0,0 +1,139 @@
+
+
+class TestMysqlIntegration(unittest.TestCase):
+    def test_span_succeeded(self):
+        mock_tracer = MockTracer()
+        connection_props = {
+            "database": "testdatabase",
+            "server_host": "testhost",
+            "server_port": 123,
+            "user": "testuser",
+        }
+        mysql_tracer = MySqlTracer(mock_tracer)
+        mock_connection = mysql_tracer.wrap_connect(
+            mock_connect, {}, connection_props
+        )
+        cursor = mock_connection.cursor()
+        cursor.execute(
+            "Test query", {"param1": "param1Value", "param2": "param2Value"}
+        )
+        span = mock_tracer.get_current_span()
+
+        self.assertIs(span.kind, trace_api.SpanKind.CLIENT)
+        self.assertEqual(span.name, "mysql.testdatabase")
+        self.assertEqual(span.attributes["component"], "mysql")
+        self.assertEqual(span.attributes["db.type"], "sql")
+        self.assertEqual(span.attributes["db.instance"], "testdatabase")
+        self.assertEqual(
+            span.attributes["db.statement"],
+            "Test query params={'param1': 'param1Value', 'param2': 'param2Value'}",
+        )
+        self.assertEqual(span.attributes["db.user"], "testuser")
+        self.assertEqual(span.attributes["peer.hostname"], "testhost")
+        self.assertEqual(span.attributes["peer.port"], 123)
+        self.assertIs(
+            span.status.canonical_code, trace_api.status.StatusCanonicalCode.OK
+        )
+
+    def test_span_failed(self):
+        mock_tracer = MockTracer()
+        mysql_tracer = MySqlTracer(mock_tracer)
+        mock_connection = mysql_tracer.wrap_connect(mock_connect, {}, {})
+        cursor = mock_connection.cursor()
+        cursor.execute("Test query", throw_exception=True)
+        span = mock_tracer.get_current_span()
+
+        self.assertEqual(span.attributes["db.statement"], "Test query")
+        self.assertIs(
+            span.status.canonical_code,
+            trace_api.status.StatusCanonicalCode.UNKNOWN,
+        )
+        self.assertEqual(span.status.description, "Test Exception")
+
+
+# pylint: disable=unused-argument
+def mock_connect(*args, **kwargs):
+    database = kwargs.get("database")
+    server_host = kwargs.get("server_host")
+    server_port = kwargs.get("server_port")
+    user = kwargs.get("user")
+    return MockMySqlConnection(database, server_port, server_host, user)
+
+
+class MockMySqlConnection:
+    def __init__(self, database, server_port, server_host, user):
+        self.database = database
+        self.server_port = server_port
+        self.server_host = server_host
+        self.user = user
+
+    # pylint: disable=no-self-use
+    def cursor(self):
+        return MockMySqlCursor()
+
+
+class MockMySqlCursor:
+    # pylint: disable=unused-argument, no-self-use
+    def execute(self, query, params=None, throw_exception=False):
+        if throw_exception:
+            raise Exception("Test Exception")
+
+
+class MockSpan:
```

Comment:
are MockSpan and MockTracer needed? I feel like we have a no-op tracer now with the API.

Reply:
I wanted to test the integration code is creating the span correctly, API tracer is no-op so it will not create anything, are you suggesting to use unittest.Mock with a side effect or some other pattern to test here?

Reply:
Using no-op tracer in tests but still using MockSpan to check all attributes are set up correctly

Reply:
could use a Mock object which takes the Tracer as as a spec: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock

Reply:
if the purpose here is to provide a general test utility for tracers, I guess I'd rather see it shared in something like opentelemetry-api. But I'm not sure if sharing test fixtures across projects is an accepted policy.

Reply:
Mocking tracer using mock library now, for Span I prefer to create my own mock class so is easier to test the integration adding correct attributes to the Span